mcs/class/corlib/System.Security.Cryptography.X509Certificates/X509Helper.cs

```diff
@@ -49,22 +49,30 @@ internal static void InstallNativeHelper (INativeCertificateHelper helper)
 				Interlocked.CompareExchange (ref nativeHelper, helper, null);
 		}
 
-#if !MOBILE
-		[SecurityPermission (SecurityAction.Demand, UnmanagedCode = true)]
-#endif
 		public static X509CertificateImpl InitFromHandle (IntPtr handle)
 		{
 #if MONO_FEATURE_APPLETLS
-			var variable = Environment.GetEnvironmentVariable ("MONO_TLS_PROVIDER");
-			bool useAppleTls = string.IsNullOrEmpty (variable) || variable == "default" || variable == "apple"; // On Platform.IsMacOS default is AppleTlsProvider
-			if (System.Environment.IsMacOS && useAppleTls)
+			if (System.Environment.IsMacOS && Environment.GetEnvironmentVariable ("MONO_TLS_PROVIDER") != "legacy")
 				return InitFromHandleApple (handle);
 #endif
-#if !MOBILE
+#if !ONLY_APPLETLS && !MOBILE
 			InitFromHandleCore (handle);
 #endif
 			throw new NotSupportedException ();
 		}
+		
+		static X509CertificateImpl Import (byte[] rawData)
+		{
+#if MONO_FEATURE_APPLETLS
+			if (System.Environment.IsMacOS && Environment.GetEnvironmentVariable ("MONO_TLS_PROVIDER") != "legacy")
+				return ImportApple (rawData);
+#endif
+#if !ONLY_APPLETLS && !MOBILE
```

Comment:
Also, we need the !MOBILE @marek-safar because of monodroid, which hit this case:

https://github.com/mono/mono/blob/master/mcs/class/corlib/System.Security.Cryptography.X509Certificates/X509Helper.cs#L85

since it is MOBILE and not MONOTOUCH || XAMMAC